File: Backend/orderservice/src/main/java/com/cts/orderservice/service/OrderService.java
package com.cts.orderservice.service;

import java.util.List;

import com.cts.orderservice.model.Order;

public interface OrderService {

	public Order createOrder(Long userId) ;
	 public List<Order> getOrdersByUserId(Long userId);
	 public Order getOrderById(Long orderId);
	
}

File: Frontend/booknest/src/app/core/services/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Order } from '../models/order';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  private baseUrl = 'http://localhost:8765/api/orders';
 
  constructor(private http: HttpClient) { }
 
  createOrder(userId: number): Observable<Order> {
return this.http.post<Order>(`${this.baseUrl}/create`, { userId });
  }
 
  getOrderById(orderId: number): Observable<Order> {
    return this.http.get<Order>(`${this.baseUrl}/${orderId}`);
  }
 
  getOrdersByUserId(userId: number): Observable<Order[]> {
    return this.http.get<Order[]>(`${this.baseUrl}/user/${userId}`);
  }
}
